### ISSUE TITLE
expireAfterSeconds for mongo indexes

### ISSUE THREAD

**Issue by snackycracky** [e2nIEE/pandahub#70]
Hello :) 

I would have expected that `expireAfterSeconds` ([mongo docs](https://www.mongodb.com/docs/manual/tutorial/expire-data/)) is configured here: https://github.com/e2nIEE/pandahub/blob/develop/pandahub/lib/mongodb_indexes.py#L8
Otherwise the data ist going to explode ;)  

Grüße

**Comment by vogt31337** [comment]
I'll encourage that, for example an Index for trafo3w is missing.

**Comment by jthurner** [comment]
@snackycracky 

> I would have expected that expireAfterSeconds ([mongo docs](https://www.mongodb.com/docs/manual/tutorial/expire-data/)) is configured here: https://github.com/e2nIEE/pandahub/blob/develop/pandahub/lib/mongodb_indexes.py#L8

Well, I would definitely NOT expect my data to just vanish after some arbitrary time :laughing:

>Otherwise the data ist going to explode ;)

Why would the data explode? It's not going to multiply by itself, maybe an easier way would be to apply some cleanup logic on inserting new data, and to use projects/networks as abstractions instead of working on the mongodb document level.

That said, if  you really need to set TTL on inserted documents for some reason, you could implement it relatively easily along these lines (untested):

```python
import inspect
import pandas as pd
from datetime import datetime, UTC
from pandahub import PandaHub
from pandahub.lib.mongodb_indexes import MONGODB_INDEXES
from pymongo import IndexModel, DESCENDING

# extend the default indices with a  TTL index on inserted_at
MY_MONGODB_INDICES = {
    element: default_indices + [IndexModel([("inserted_at", DESCENDING)], expireAfterSeconds=30)]
    for element, default_indices in MONGODB_INDEXES.items()
}

class ForgetfulPandaHub(PandaHub):
    """PandaHub subclass initialized with custom indices and timestamps added to net df's in write_network_to_db."""
    def __init__(self, *args, **kwargs) -> None:
        super().__init__(*args, mongodb_indexes=MY_MONGODB_INDICES, **kwargs)

    def write_network_to_db(self, *args, **kwargs) -> dict:
        inserted_at = datetime(tzinfo=UTC)
        fn_args = inspect.signature(self.write_network_to_db).bind(*args, **kwargs)
        fn_args.apply_defaults()
        for element in fn_args.arguments["net"].keys():
            if isinstance(fn_args.arguments["net"][element], pd.DataFrame):
                fn_args.arguments["net"][element]["inserted_at"] = inserted_at
        return super().write_network_to_db(*fn_args.args,**fn_args.kwargs)
```

.. then use `ForgetFulPandahub()` as usual and all your data will be gone 30s after you inserted it.

More generally, this would be the recommended way to add custom mongodb indexes to support your own queries not covered by the defaults - extend the provided ones and pass them to PandaHub on instantiation.

**Comment by jthurner** [comment]
@vogt31337 

>  Index for trafo3w is missing.

True, added default indices for trafo3w and some other elements in #102.
